streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignorTest.java

```diff
@@ -2360,4 +2662,27 @@ private static Map<TaskId, Long> getTaskEndOffsetSums(final Collection<TaskId> a
         return allStatefulTasks.stream().collect(Collectors.toMap(t -> t, t -> Long.MAX_VALUE));
     }
 
+    private static List<Map<String, List<TopicPartitionInfo>>> getTopicPartitionInfo(final int replicaCount, final List<Set<String>> topicsList) {
+        final List<KeyValue<Node, List<Node>>> nodes = asList(
+            KeyValue.pair(NODE_0, asList(REPLICA_0)),
+            KeyValue.pair(NODE_1, asList(REPLICA_1)),
+            KeyValue.pair(NODE_2, asList(REPLICA_2)),
+            KeyValue.pair(NODE_3, asList(REPLICA_3))
+        );
+
+        final List<Map<String, List<TopicPartitionInfo>>> ret = new ArrayList<>();
+        for (final Set<String> topics : topicsList) {
+            ret.add(new HashMap<>());
+            for (final String topic : topics) {
+                final List<TopicPartitionInfo> topicPartitionInfoList = new ArrayList<>();
+                ret.get(ret.size() - 1).put(topic, topicPartitionInfoList);
```

Comment:
Should we add a local variable instead of getting the map from the list?
```
final Map topicToPartitionInfo = new HashMap();
for (....) {
  ...
  topicToPartitionInfo.put(topic, topicPartitionInfoList);
}
ret.add(topicToPartitionInfo);
```